### DISCUSSION TITLE
Attribution for contributions

### DISCUSSION THREAD

**Discussion by tino097** [ckan/ckan#6608] [Ideas]
There was already mentioned on the tech team call but its good to set as discussion for the community. It was asked few times on different PR's and on the gitter channel. 

This could bebenefitial to get some of the CKAN core extensions more activly maintained and some third party extensions to get more synced with the current release. 

The one option bringed by @Zharktas was as their release [contributors](https://github.com/vrk-kpa/opendata/releases/tag/v3.44.0) section which we could adopt to our releases.

Other option were mentioned as Rust's [contributors](https://thanks.rust-lang.org/rust/1.57.0/) list @smotornyuk 

Happy to hear more experiences from other projects

**Comment by jqnatividad** [comment]
We should also encourage attributions for contributions that are funded/sponsored by organizations.

When we started Ontodia 10 years ago, we made sure to always try to contribute back our customizations, as we were a very small team and stood on the shoulders of the community.  When Ontodia was acquired by OpenGov in 2016, we increased our contributions and even managed to sponsor some core work.

We justified it to management several ways:
* You're saving money. We're sponsoring very productive CKAN experts that don't have to learn CKAN on the job, you don't have to pay them a full salary (which was very expensive in Silicon Valley), they're mission-driven and won't jump ship two years later (average stay of engineers before they leave for greener pa$ture$).
* You gain visibility/credibility in the community which helps with sales/marketing.
* It becomes a selling point for your customers as enterprises embrace open source.
* It's a differentiator against your competitors with proprietary solutions that lock you in.

We also tell our customers that it is in their enlightened self-interest to contribute, as:
* if the contribution is accepted, it's maintained by the project and you don't end up with legacy code, bitrot issues
* we will call out your contribution as well

Now that we started datHere, we're doubling down our open-source contributions even more now that we're in startup mode again, as it just makes sense:
* you get to work with self-selecting, mission-driven individuals/organizations who want to collaborate not solely because they're getting paid to do it, they actually like what they're working on.
* you get to learn & benefit from their contributions as well and you don't have to build everything yourself

Last year for example, we did some work for a hedge fund and they were very particular about not being called out for various reasons, but the client sponsor was more than happy to allow us to contribute a lot of the work we did back to the project.

It even allowed me to start maintaining a new open source project - [qsv](https://github.com/jqnatividad/qsv), as we were using the "paused" [xsv](https://github.com/BurntSushi/xsv) project (from which qsv was forked) heavily in their data pipelines for their CKAN catalog.

**Reply by kepol** [reply]
I'm a very active Drupal contributor and have been for years.

In case you aren't aware, Drupal has a very robust (though bespoke) contribution credit system to attribute contributions to individuals who are volunteering their own time or who are doing the contributions on behalf of organizations. They can attribute their work to multiple organizations. For example, if they are working at a web agency on a client project, both of these organizations can be giving contribution credit.

This modal helps provides a visible incentive to organizations to increase their contribution. There is a "leader board" of sorts at:

https://www.drupal.org/organizations

where organizations are ranked based on contributions. Not all contributions are weighted equally. For example, Drupal core contribution is weighted much higher than for a contributed module with low usage.

For smaller projects, this bespoke approach is not practical and even Drupal will be moving to using GitLab for contribution credits at some point in the future. But, it shows the power of giving organizations credit for sponsoring individual contribution.

I haven't reviewed the resources above yet but wanted to quickly chime in on this to echo what @jqnatividad wrote above:

> We should also encourage attributions for contributions that are funded/sponsored by organizations.
